File: Advanced C#/Data-Structures/StacksAndQueues/StacksAndQueues/Stack.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StacksAndQueues
{
    public class Stack
    {
        public Node Top { get; set; }

        public void Push(string value)
        {
            // Create a new node
            Node node = new Node(value);
            node.Next = Top;
            Top = node;

        }

        public string Pop()
        {
            if (Top == null)
            {
                throw new Exception("Empty stack");
            }
            string result = Top.Value;
            Top = Top.Next;
            return result;
        }

        public string Peek()
        {
            if (Top == null)
            {
                throw new Exception("Empty stack");
            }
            return Top.Value;
        }
    }
}

File: Advanced-Javascript/data-structure/StacksandQueues/stack.js
'use strict';

const Node = require('../linked-list/node.js');

class Stack{
    constructor(){
        this.top = null;
    }

    push(value){
        let node = new Node(value);
        node.next = this.top;
        this.top = node;
        return this.top.value;
    }

    pop(){
        if(this.top === null){
            return null;
        }
        let result = this.top.value;
        this.top = this.top.next;
        return result;
    }

    peek(){
        if(this.top === null){
            return null;
        }

        return this.top.value;
    }
}

module.exports = Stack;